test/suite/test_prepare01.py

```diff
@@ -0,0 +1,177 @@
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# test_prepare01.py
+#    Transactions: basic functionality with prepare
+class test_prepare01(wttest.WiredTigerTestCase):
+    nentries = 1000
+    scenarios = make_scenarios([
+        ('col-f', dict(uri='file:text_txn01',key_format='r',value_format='S')),
+        ('col-t', dict(uri='table:text_txn01',key_format='r',value_format='S')),
+        ('fix-f', dict(uri='file:text_txn01',key_format='r',value_format='8t')),
+        ('fix-t', dict(uri='table:text_txn01',key_format='r',value_format='8t')),
+        ('row-f', dict(uri='file:text_txn01',key_format='S',value_format='S')),
+        ('row-t', dict(uri='table:text_txn01',key_format='S',value_format='S')),
+    ])
+
+    # Return the number of records visible to the cursor.
+    def cursor_count(self, cursor):
+        count = 0
+        # Column-store appends result in phantoms, ignore records unless they
+        # have our flag value.
+        for r in cursor:
+            if self.value_format == 'S' or cursor.get_value() == 0xab:
+                count += 1
+        return count
+
+    # Checkpoint the database and assert the number of records visible to the
+    # checkpoint matches the expected value.
+    def check_checkpoint(self, expected):
+        s = self.conn.open_session()
+        s.checkpoint("name=test")
+        cursor = s.open_cursor(self.uri, None, "checkpoint=test")
+        self.assertEqual(self.cursor_count(cursor), expected)
+        s.close()
+
+    # Open a cursor with snapshot isolation, and assert the number of records
+    # visible to the cursor matches the expected value.
+    def check_txn_cursor(self, level, expected):
+        s = self.conn.open_session()
+        cursor = s.open_cursor(self.uri, None)
+        s.begin_transaction(level)
+        self.assertEqual(self.cursor_count(cursor), expected)
+        s.close()
+
+    # Open a session with snapshot isolation, and assert the number of records
+    # visible to the cursor matches the expected value.
+    def check_txn_session(self, level, expected):
+        s = self.conn.open_session(level)
+        cursor = s.open_cursor(self.uri, None)
+        s.begin_transaction()
+        self.assertEqual(self.cursor_count(cursor), expected)
+        s.close()
+
+    def check(self, cursor, committed, total):
```

Comment:
In this test function check() are we testing anything additional to what is already tested in test_txn01.py?

Reply:
This check is to test that all are visible from with in the same cursor different from others and as per comment.